components/dashboards-web-component/src/utils/WidgetPubSubHub.js

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+export default class WidgetPubSubHub {
+    constructor(glEventHub) {
+        this.glEventHub = glEventHub;
+        this.lastPublishedMessages = {};
+    }
+
+    /**
+     * Publishes a message to subscribers.
+     * @param {string} publisherId ID of the message publisher
+     * @param {object} message publishing message
+     */
+    publish(publisherId, message) {
+        this.lastPublishedMessages[publisherId] = message;
+        this.glEventHub.emit(publisherId, message);
+    }
+
+    /**
+     * Subscribes to messages.
+     * @param {string} publisherId ID of the message publisher that needs to listen to
+     * @param {function(message)} listenerCallback The listener function that should be invoked when a message received
+     * @param {object} [listenerThisContext] The value of the <code>this</code> pointer in the listener function
+     */
+    subscribe(publisherId, listenerCallback, listenerThisContext) {
+        const lastPublishedMessage = this.lastPublishedMessages[publisherId];
+        if (lastPublishedMessage) {
+            listenerCallback.call(listenerThisContext, lastPublishedMessage);
```

Comment:
Last published message needs to be checked only once. After that subscriber can receive messages from the publisher directly.

Reply:
A subscriber widget calls this function only **once** for a particular publisher. So this line gets executed only once. After directly wired with the `glEventHub` (see line 46), this function doesn't get called.